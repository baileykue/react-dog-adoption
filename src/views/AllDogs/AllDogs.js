import React, { useState, useEffect } from 'react';
import { fetchDogs } from '../../services/dogs';
import DogList from '../../components/DogList/DogList';

export default function AllDogs() {
  const [dogs, setDogs] = useState([]);

  useEffect(() => {
    const fetchData = async () => {
      const data = await fetchDogs();
      setDogs(data);
    };
    fetchData();
  }, []);

  return (
    <div className="dogs">
      <DogList {...dogs} />
    </div>
  );
}
