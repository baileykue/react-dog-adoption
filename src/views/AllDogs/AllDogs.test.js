import AllDogs from './AllDogs';
import { render, screen } from '@testing-library/react';

test.skip('', () => {
  const { conatiner } = render(<AllDogs />);
  const nameElement = screen.getByText('Barton');
  expect(nameElement).toBeInTheDocument();
});
