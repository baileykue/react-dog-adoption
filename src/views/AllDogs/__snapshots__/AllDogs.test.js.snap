// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<AllDogs/> should render the entire list of dogs 1`] = `
<div>
  <div
    class="home"
  >
    <h1>
      Dogs Available For Adoption
    </h1>
    <div
      class="dogs"
    >
      <a
        href="/dogs/1"
      >
        <div
          class="card"
        >
          <img
            src="https://placedog.net/1000/563?id=19"
          />
          <h2>
            Stevie
          </h2>
          <p>
            Hi! I am a(n) 
            0
             year(s) old 
            White Labrador
            . I cannot wait to meet you!
          </p>
        </div>
      </a>
      <a
        href="/dogs/5"
      >
        <div
          class="card"
        >
          <img
            src="https://placedog.net/1000/563?id=135"
          />
          <h2>
            Stripe
          </h2>
          <p>
            Hi! I am a(n) 
            3
             year(s) old 
            Dalmatian
            . I cannot wait to meet you!
          </p>
        </div>
      </a>
      <a
        href="/dogs/6"
      >
        <div
          class="card"
        >
          <img
            src="https://placedog.net/1000/563?id=54"
          />
          <h2>
            Bailey
          </h2>
          <p>
            Hi! I am a(n) 
            2
             year(s) old 
            mutt
            . I cannot wait to meet you!
          </p>
        </div>
      </a>
      <a
        href="/dogs/7"
      >
        <div
          class="card"
        >
          <img
            src="https://placedog.net/1000/563?id=64"
          />
          <h2>
            Squiggle
          </h2>
          <p>
            Hi! I am a(n) 
            3
             year(s) old 
            mutt
            . I cannot wait to meet you!
          </p>
        </div>
      </a>
      <a
        href="/dogs/9"
      >
        <div
          class="card"
        >
          <img
            src="https://placedog.net/1000/563?id=109"
          />
          <h2>
            Jasper
          </h2>
          <p>
            Hi! I am a(n) 
            0
             year(s) old 
            Husky
            . I cannot wait to meet you!
          </p>
        </div>
      </a>
      <a
        href="/dogs/10"
      >
        <div
          class="card"
        >
          <img
            src="https://placedog.net/1000/563?id=3"
          />
          <h2>
            King
          </h2>
          <p>
            Hi! I am a(n) 
            2
             year(s) old 
            Golden Retriever
            . I cannot wait to meet you!
          </p>
        </div>
      </a>
      <a
        href="/dogs/3"
      >
        <div
          class="card"
        >
          <img
            src="https://placedog.net/1000/563?id=51"
          />
          <h2>
            Frenchie
          </h2>
          <p>
            Hi! I am a(n) 
            1
             year(s) old 
            French Bulldog
            . I cannot wait to meet you!
          </p>
        </div>
      </a>
    </div>
  </div>
</div>
`;
